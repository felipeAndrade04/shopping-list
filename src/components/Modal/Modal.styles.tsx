import { Modal } from 'react-native';
import styled from 'styled-components/native';

export const Container = styled(Modal)``;

export const Overlay = styled.TouchableOpacity.attrs(() => ({
  activeOpacity: 1,
}))`
  width: 100%;
  height: 100%;
  background: ${({ theme }) => `${theme.colors.overlay}aa`};
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 16px;
`;

export const Content = styled.TouchableOpacity.attrs(() => ({
  activeOpacity: 1,
}))`
  width: 100%;
  padding: 12px;
  background: ${({ theme }) => theme.colors.gray[100]};
  border-radius: 16px;
`;
