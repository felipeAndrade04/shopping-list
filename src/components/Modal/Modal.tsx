import React from 'react';
import * as S from './Modal.styles';
import { ModalProps } from './Modal.types';

export function Modal({ children, show, close }: ModalProps) {
  return (
    <S.Container
      animationType="fade"
      transparent={true}
      visible={show}
      onRequestClose={() => {
        close();
      }}
    >
      <S.Overlay onPress={() => close()} testID="overlay">
        <S.Content>{children}</S.Content>
      </S.Overlay>
    </S.Container>
  );
}
