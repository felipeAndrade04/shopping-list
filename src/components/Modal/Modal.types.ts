import { ReactNode } from 'react';

export interface ModalProps {
  show: boolean;
  children: ReactNode;
  close: () => void;
}
