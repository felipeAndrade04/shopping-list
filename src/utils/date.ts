const months = ['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set', 'out', 'nov', 'dez'];

export function formatDate(date: Date) {
  const day = date.getDay();
  const month = date.getMonth();

  return `${day} de ${months[month]}`;
}
