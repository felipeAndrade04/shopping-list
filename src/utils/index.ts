export * from './delay';
export * from './date';
