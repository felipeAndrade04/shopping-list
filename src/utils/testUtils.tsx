import React from 'react';
import { authReducer } from '@app/store';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/native';
import * as theme from '@app/theme';
import { configureStore } from '@reduxjs/toolkit';

let store = configureStore({
  reducer: {
    auth: authReducer,
  },
});

export const wrapper = ({ children }: { children: ReactNode }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>{children}</Provider>
  </ThemeProvider>
);

export function resetStore() {
  store = configureStore({
    reducer: {
      auth: authReducer,
    },
  });
}
