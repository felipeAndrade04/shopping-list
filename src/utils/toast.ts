import EventManager from './EvenManager';

export type ToastType = 'success' | 'info' | 'error';

export interface Toast {
  id?: string;
  type: ToastType;
  text: string;
  duration?: number;
}

export const toastEventManager = new EventManager();

export default function toast({ type, text, duration }: Toast) {
  toastEventManager.emit<Toast>('addtoast', {
    id: new Date().getTime().toString(),
    type,
    text,
    duration,
  });
}
