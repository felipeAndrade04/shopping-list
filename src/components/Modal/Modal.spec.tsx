import React from 'react';
import { Text } from 'react-native';
import { fireEvent, render } from '@testing-library/react-native';
import { wrapper } from '@app/utils/testUtils';
import { Modal } from './Modal';

describe('<Modal />', () => {
  it('Should render correctly', () => {
    const tree = render(
      <Modal show={true} close={() => jest.fn}>
        <Text>Test</Text>
      </Modal>,
      { wrapper }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should show content on screen', () => {
    const content = 'Modal Text';

    const { queryByText } = render(
      <Modal show={true} close={() => jest.fn}>
        <Text>{content}</Text>
      </Modal>,
      { wrapper }
    );

    expect(queryByText(content)).toBeTruthy();
  });

  it('Should not show content on screen', () => {
    const content = 'Modal Text';

    const { queryByText } = render(
      <Modal show={false} close={() => jest.fn}>
        <Text>{content}</Text>
      </Modal>,
      { wrapper }
    );

    expect(queryByText(content)).toBeFalsy();
  });

  it('Shold close modal', () => {
    const content = 'Modal Text';

    let show = true;

    const { queryByTestId } = render(
      <Modal show={show} close={() => (show = false)}>
        <Text>{content}</Text>
      </Modal>,
      { wrapper }
    );

    expect(show).toBeTruthy();

    const overlay = queryByTestId('overlay');
    fireEvent.press(overlay);

    expect(show).toBeFalsy();
  });

  it('Shold call close function', () => {
    const content = 'Modal Text';

    const close = jest.fn();

    const { queryByTestId } = render(
      <Modal show={true} close={close}>
        <Text>{content}</Text>
      </Modal>,
      { wrapper }
    );

    const overlay = queryByTestId('overlay');
    fireEvent.press(overlay);

    expect(close).toBeCalledTimes(1);
  });
});
