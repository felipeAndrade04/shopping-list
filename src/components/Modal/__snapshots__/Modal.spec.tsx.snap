// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Should render correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  style={
    [
      {},
    ]
  }
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#363F5Faa",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": "100%",
      }
    }
    testID="overlay"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#F4F4F4",
          "borderRadius": 16,
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text>
        Test
      </Text>
    </View>
  </View>
</Modal>
`;
